packages/dx-react-scheduler/src/plugins/confirmation-dialog.tsx

```diff
@@ -0,0 +1,149 @@
+  buttonComponent, ignoreDelete, ignoreCancel,
+}) => {
+  const getMessage = getMessagesFormatter({ ...defaultMessages, ...messages });
+  const modalContainer = React.useRef();
+
+  const [isOpen, setIsOpen] = React.useState(false);
+  const [actionType, setActionType] = React.useState('');
+  const [hideActionName, setHideActionName] = React.useState('');
+  const [appointmentData, setAppointmentData] = React.useState({});
+
+  const toggleIsOpen = React.useCallback(() => {
+    setIsOpen(!isOpen);
+  }, [isOpen, setIsOpen]);
+
+  const confirmCancelChanges = React.useCallback((hideAction) => {
+    toggleIsOpen();
+    setHideActionName(hideAction);
+    setActionType(ACTION_TYPES.CANCEL);
+  }, [toggleIsOpen, setHideActionName, setActionType]);
+
+  const confirmDelete = React.useCallback(({
+    hideActionName: hideAction, appointmentData: changedAppointment,
+  }) => {
+    toggleIsOpen();
+    setHideActionName(hideAction);
+    setActionType(ACTION_TYPES.DELETE);
+    setAppointmentData(changedAppointment);
+  }, [toggleIsOpen, setHideActionName, setActionType, setAppointmentData]);
+
+  const confirmAction = React.useCallback((
+    isNewAppointment, hideEditor, stopEditAppointment, finishDeleteAppointment,
+    cancelAddedAppointment, cancelChangedAppointment,
+  ) => () => {
+    hideEditor();
+    toggleIsOpen();
+    if (isNewAppointment) {
+      callActionIfExists(cancelAddedAppointment, appointmentData);
+    } else {
+      callActionIfExists(stopEditAppointment, appointmentData);
+      callActionIfExists(cancelChangedAppointment, appointmentData);
+    }
+    if (actionType === ACTION_TYPES.DELETE && finishDeleteAppointment) {
+      finishDeleteAppointment(appointmentData);
+    }
+  }, [toggleIsOpen, actionType, appointmentData]);
+
+  return (
+    <Plugin
+      name="ConfirmationDialog"
+      dependencies={pluginDependencies}
+    >
+      {!ignoreCancel &&
+        <Action name="openCancelConfirmationDialog" action={confirmCancelChanges} />
+      }
+      {!ignoreDelete &&
+        <Action name="openDeleteConfirmationDialog" action={confirmDelete} />
+      }
+
+      <Template name="schedulerRoot">
+        <TemplatePlaceholder />
+        <Container ref={modalContainer} />
+        <TemplatePlaceholder name="confirmationDialogOverlay" />
+      </Template>
+
+      <Template name="confirmationDialogOverlay">
+        <TemplateConnector>
+          {({
+            editingAppointment,
+          }, actions) => {
+            const handleConfirm = confirmAction(
+              !editingAppointment, actions[hideActionName], actions.stopEditAppointment,
+              actions.finishDeleteAppointment, actions.cancelAddedAppointment,
+              actions.cancelChangedAppointment,
+            );
+
+            return (
+              <Overlay
+                target={modalContainer}
+                visible={isOpen}
+                onHide={() => isOpen && toggleIsOpen()}
```

Comment:
`() => isOpen && toggleIsOpen()`

Users can call this action only if a button is visible, isn't it?
```
onHide={toggleIsOpen}
```